### PR TITLE
Add static local functions proposal

### PR DESCRIPTION
**Pull request by cston**
Add proposal covering the C#8 feature.

### CODE REVIEW

File: proposals/static-local-functions.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+A local function declared `static` cannot capture state from the enclosing scope.
+As a result, locals, parameters, and `this` from the enclosing scope are not available within a `static` local function.
+
+A `static` local function cannot reference instance members from an implicit or explicit `this` or `base` reference.
+
+A `static` local function may reference `static` members from the enclosing scope.
+
+A `static` local function may reference `constant` definitions from the enclosing scope.
+
+`nameof()` in a `static` local function cannot reference state from the enclosing scope.
```
Comment (CyrusNajmabadi) [inline proposals/static-local-functions.md]: that seems super strange.  why?  nameof is just a constant, so it should be fine, right?

**Comment by jcouv** [comment]
This doc should probably be under the chsarp8 folder

**Comment by BillWagner** [comment]
Once this is merged, I'll update the TOC in docs to include it with the other C# 8 proposals.

**Comment by cston** [comment]
@jaredpar, @jcouv, please review.

**Pull request merged**